src/main/kotlin/controller/ImageController.kt

```diff
@@ -36,20 +40,31 @@ class ImageController : Controller() {
         activeImage.image.set(im)
     }
 
-    fun applyFilter(op: FilterOperation) {
+    fun applyFilter(op: FilterOperation, isChecked: Boolean) {
         val raw = resultImage.image
         val image = WritableImage(raw.pixelReader, raw.width.toInt(), raw.height.toInt())
         when (op) {
             FilterOperation.GREYSCALE -> BasicFilter.greyscaleFilter(image)
             FilterOperation.INVERSE_COLOR -> BasicFilter.inverseColorFilter(image)
+            FilterOperation.MIRROR -> BasicFilter.mirrorFilter(image)
         }
         resultImage.image = image
         isRaw = false
-        activeImage.image.set(resultImage.image)
+        if (isChecked) activeImage.image.set(resultImage.image) else activeImage.image.set(rawImage.image)
```

Comment:
As said in the group chat, image comparison is already implemented. If you notice the variable `isRaw`, it is doing the job

Reply:
I wrote this code for toggling between images when the checkbox on the left checked and unchecked, not aiming for the comparison, though quite buggy in this situation